src/librustdoc/clean/utils.rs

```diff
@@ -398,10 +398,10 @@ crate fn register_res(cx: &mut DocContext<'_>, res: Res) -> DefId {
             i,
         ) => (i, kind.into()),
         // This is part of a trait definition; document the trait.
-        Res::SelfTy(Some(trait_def_id), _) => (trait_def_id, ItemType::Trait),
+        Res::SelfTy { trait_: Some(trait_def_id), alias_to: _ } => (trait_def_id, ItemType::Trait),
         // This is an inherent impl; it doesn't have its own page.
-        Res::SelfTy(None, Some((impl_def_id, _))) => return impl_def_id,
-        Res::SelfTy(None, None)
+        Res::SelfTy { trait_: None, alias_to: Some((impl_def_id, _)) } => return impl_def_id,
+        Res::SelfTy { trait_: None, alias_to: None }
```

Comment:
I am not familiar with rustdoc so am not sure if this is actually a bug but these comments seem wrong :sweat_smile: 
the `trait_: Some(trait)` case is not _just_ for trait definitions and is also there in `impl Trait for Foo` blocks
the `trait_: None, alias_to: Some(..)` is not _just_ for inherent impls and is also there in struct defs (`struct Foo(Box<Self>)`)

Reply:
If you can, please go ahead and update these comments so they're correct :)

Reply:
i think it makes sense to add a fixme to potentially change this to
```rust
Res::SelfTy { trait_: Some(trait_def_id), alias_to: None } => (trait_def_id, ItemType::Trait),
```
i think the mention of `impl Send for [u8]` might be caused by this https://doc.rust-lang.org/nightly/std/marker/trait.Send.html#impl-Send-80

but i don't want to add functionality changes to rustdoc in this PR